src/java.base/share/classes/java/io/RandomAccessFile.java

```diff
@@ -57,6 +57,8 @@
  * than {@code EOFException} is thrown. In particular, an
  * {@code IOException} may be thrown if the stream has been closed.
  *
+ * @implNote This class is not thread safe.
```

Comment:
I don't think this should be an implNote. Maybe for a different issue/PR is that the class description probably needs a paragraph to say that a RAF is not thread safe and requires appropriate synchronization when used by concurrent threads. This leads to discussion about a FileChannel obtained from a RAF as FileChannel defines read/write methods that support cases where threads need to access different parts of a file at the same time. There is a lot more to say on this topic and probably best to keep it separate.

Reply:
Nit: Also I think it needs hyphenation: `thread-safe`.